### DISCUSSION TITLE
Redirects in vercel.json with dynamic patterns

### DISCUSSION THREAD

**Discussion by janniks** [vercel/community#1081] [Help] [answered]
Can somebody help out with patterns used in vercel.json redirects?
```
{
 "source": "/modules/:path((?!_stacks_).*)",
 "destination": "/modules/_stacks_:path*",     
 "permanent": false
}
```
This works and `path` is dynamic (can be anything), if "_stacks" is missing, it's added.
Now i would want to additionally make this rule work for every path-prefix (not just `/modules/`) is it possible to make the "modules" part dynamic as well?
Thanks in advance 🙏

**Comment by amyegan** [comment, marked as answer]
Good question @janniks! If you change the config to
```
{
 "source": "/:first/:path((?!_stacks_).*)",
 "destination": "/modules/_stacks_:path*",     
 "permanent": false
}
```
then it'll redirect `/one/two` to `/modules/_stacks_two` and `/one/two/three` to `/modules/_stacks_two/three`

**Reply by janniks** [reply]
Hmm, so there's not possible single config redirect for this?
- turn `/one/two` to `/one/_prefix_two`
- turn `/foo/two` to `/foo/_prefix_two`

**Reply by amyegan** [reply]
I'm sorry, I misunderstood what you wanted. You can redirect as you described if you change the config to something like this:
```
{
 "source": "/:first/:path((?!_stacks_).*)",
 "destination": "/:first/_stacks_:path*"
}
```
